storage/postgres/security.go

```diff
@@ -17,17 +17,53 @@
 package postgres
 
 import (
+	"context"
 	"fmt"
+	"sync"
 	"time"
 
 	"github.com/Peripli/service-manager/security"
 	"github.com/jmoiron/sqlx"
 	"github.com/sirupsen/logrus"
 )
 
+const securityLockIndex = 111
+
 type securityStorage struct {
 	db            *sqlx.DB
 	encryptionKey []byte
+	isLocked      bool
+	mutex         *sync.Mutex
+}
+
+// Lock acquires a database lock so that only one process can manipulate the encryption key.
+// Returns an error if the process has already acquired the lock
+func (s *securityStorage) Lock(ctx context.Context) error {
+	s.mutex.Lock()
```

Comment:
is this the correct lock ( e.x. this or RWLock )

Reply:
RWMutex provides only R(W)Lock and R(W)Unlock for these explicit situations. RWMutext.Lock/Unlock is the same as Mutex.Lock/Unlock
Since this is both reading and writing the value, there's no difference between RWMutex.Lock and Mutex.Lock